brut.apktool/apktool-lib/src/main/java/brut/androlib/res/decoder/ARSCDecoder.java

```diff
@@ -232,17 +232,17 @@ private ResBagValue readComplexEntry() throws IOException,
 
         ResValueFactory factory = mPkg.getValueFactory();
         Duo<Integer, ResScalarValue>[] items = new Duo[count];
-        ResValue resValue;
+        ResIntBasedValue resValue;
         int resId;
 
         for (int i = 0; i < count; i++) {
             resId = mIn.readInt();
             resValue = readValue();
 
-            try {
```

Comment:
I like your version better, my local testing showed no regression with the apks that I patched the original issue for, and the non exception catch should prove to be quicker. Everything looks good on my end.

Thanks for the quick pull request. I'll get this merged into master.